### PR TITLE
Bug 1928147: Respect DHCP option 119 for on-prem

### PR DESCRIPTION
**Pull request by darkmuggle**
On-prem scripts were swallowing IPV4 domain searches.

### CODE REVIEW

**Comment by darkmuggle** [comment]
@miabbott @ashcrow for the MCO team to ack this, we'll need an RHCOSian (can that be a term?) to place the stamp of approval on this.

**Comment by ashcrow** [comment]
> RHCOSian

Sounds like an :alien: ... but sure :smile:

**Comment by miabbott** [comment]
> @miabbott @ashcrow for the MCO team to ack this, we'll need an RHCOSian (can that be a term?) to place the stamp of approval on this.

Seems sane, though I'd like to see someone from RHV/oVirt give a thumps up too.  cc: @Gal-Zaidman

**Comment by kikisdeliveryservice** [comment]
PTAL (also adding holding until some baremetal&friends review)

/assign @cybertron @Gal-Zaidman
/hold

File: templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml
Comment (cybertron) [inline templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml:53]: While we're at it, should we also do IP6_DOMAINS?

Could also make this reference "set -u" safe by doing ${IP4_DOMAINS:-}, but this script is full of unsafe variable references so I don't know that it actually matters.

Comment (darkmuggle) [reply templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml]: In regards to `$IP6_DOMAINS`  I've update and repushed.

However, in terms of script saftey, `set -u` would be wise, however:
- we cannot guarantee that NW will set those values
- improve the safety of the script is likely to be disruptive -- the script was not written with safety in mind and turning it on will likely introduce unpredictable behavior changes and is likely an initiative that should be driven by the on-prem platform teams.

Comment (cybertron) [reply templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml]: Yeah, I wouldn't want to enable set -u in this patch.

**Comment by cybertron** [comment]
lgtm pending ci

**Comment by cybertron** [comment]
/lgtm
/retest
/test e2e-openstack
/test e2e-ovirt
/test e2e-vsphere

Passed metal-ipi so I would expect this to be good to go. Apologies for not triggering the other on-prem platforms the last time I looked at this.

**Review by yuqi-zhang — APPROVED** [review]

**Comment by Gal-Zaidman** [comment]
> PTAL (also adding holding until some baremetal&friends review)
> 
> /assign @cybertron @Gal-Zaidman
> /hold

@kikisdeliveryservice can you remove the hold? don't know if you are waiting on another review :)

**Pull request merged**
